src/calculateAverage.js

```diff
@@ -37,7 +37,71 @@ function applyCalculateAverage() {
    * @param {Object} options - optional
    */
   [].__proto__.calculateAverage = function(options) {
-    // write code here
+    let sumOfValues = 0;
+    let countNotExistValues = 0;
+    let isAtLeastOneProp = false;
+    const processingAccumulator = (callback) => {
```

Comment:
Code looks broken and hard to read, because you create this function here and use at the end
Do you really need it? If you want will be better in this case to create `createReducer` function, which return reducer depend on params then use it once